src/Cli/AzureCliLogin.ts

```diff
@@ -95,11 +81,114 @@ export class AzureCliLogin {
         console.log(`Done registering cloud: "${this.loginConfig.environment}"`)
     }
 
+    async loginWithSecret() {
+        if (!(this.loginConfig.servicePrincipalId && this.loginConfig.tenantId && this.loginConfig.servicePrincipalKey) || this.isSuccess) {
+            return;
+        }
+        console.log('Attempting az cli login by using service principal with secret...\nNote: Azure/login action also supports OIDC login mechanism. If you want to use OIDC login, please do not input ClientSecret. Refer https://github.com/azure/login#configure-a-service-principal-with-a-federated-credential-to-use-oidc-based-authentication for more details.');
+        let commonArgs = ["--service-principal",
+            "-u", this.loginConfig.servicePrincipalId,
+            "--tenant", this.loginConfig.tenantId,
+            "-p", this.loginConfig.servicePrincipalKey
+        ];
+        if (this.loginConfig.allowNoSubscriptionsLogin) {
+            commonArgs.push("--allow-no-subscriptions");
+        }
+        try {
+            await this.executeAzCliCommand('login', commonArgs, true, this.loginOptions);
+            await this.setSubscription();
+            this.isSuccess = true;
+            console.log('Az cli login succeed by using service principal with secret.');

```

Comment:
Extract a new method for login, the start line is "try".
The new method will accept commonArgs and call setSubscription.